services/blockchain-indexer/shared/indexer/transactionProcessor/dpos/voteDelegate.js

```diff
@@ -13,31 +13,101 @@
  * Removal or modification of this copyright notice is prohibited.
  *
  */
+const BluebirdPromise = require('bluebird');
+
 const {
 	Logger,
 	MySQL: { getTableInstance },
 } = require('lisk-service-framework');
+
+const {
+	getBase32AddressFromPublicKey,
+	getBase32AddressFromHex,
+} = require('../../../utils/accountUtils');
 const config = require('../../../../config');
 
 const logger = Logger();
 
 const MYSQL_ENDPOINT = config.endpoints.mysql;
-const transactionsIndexSchema = require('../../../database/schema/transactions');
+const votesIndexSchema = require('../../../database/schema/votes');
+const votesAggregateIndexSchema = require('../../../database/schema/votesAggregate');
 
-const getTransactionsIndex = () => getTableInstance('transactions', transactionsIndexSchema, MYSQL_ENDPOINT);
+const getVotesIndex = () => getTableInstance('votes', votesIndexSchema, MYSQL_ENDPOINT);
+const getVotesAggregateIndex = () => getTableInstance('votes_aggregate', votesAggregateIndexSchema, MYSQL_ENDPOINT);
 
 // Command specific constants
 const commandID = 1;
 const commandName = 'voteDelegate';
 
+const getVoteIndexingInfo = async (tx) => {
+	const votesDB = await getVotesIndex();
+	const votesToAggregateArray = [];
+	const votesMultiArray = tx.params.votes.map(async vote => {
+		const voteEntry = {};
+
+		voteEntry.sentAddress = getBase32AddressFromPublicKey(tx.senderPublicKey);
+		voteEntry.receivedAddress = getBase32AddressFromHex(vote.delegateAddress);
+		voteEntry.amount = vote.amount;
+
+		const [row] = await votesDB.find({
+			id: tx.id,
+			receivedAddress: voteEntry.receivedAddress,
+			limit: 1,
+		}, ['isAggregated']);
+		if (!row || !row.isAggregated) {
+			votesToAggregateArray.push({
+				amount: BigInt(vote.amount),
+				id: voteEntry.receivedAddress.concat(voteEntry.sentAddress),
+				voteObject: {
+					...voteEntry,
+					id: voteEntry.receivedAddress.concat(voteEntry.sentAddress),
+				},
```

Comment:
Please adjust this logic as per the latests changes corresponding to the `votes` index. Also, please update the `votes_aggregate` index to use the composite primary key: `['receivedAddress', 'sentAddress']`.